src/masked-input/masked-input.tsx

```diff
@@ -33,9 +33,7 @@ const Component = (
 ) => {
     const isEmptyReadOnlyState = readOnly && isEmpty(value);
     const [isMasked, setIsMasked] = useState(!disableMask);
-    const [updatedValue, setUpdatedValue] = useState(
```

Comment:
since the `updatedValue`'s length is used for checking whether masking / mask icon should be disabled, the updatedValue shouldn't be set to "-" when the value is empty